File: sieve of eratosthenes/c++/sieve_of_eratosthenes.cpp
#include<iostream>
using namespace std;

bool* sieve_of_eratosthenes(int n) {
    bool* tab = new bool[n+1]{0};
    for(int p=2; (p*p)<=n; p++) {
        if(!tab[p]) {
            for(int i=p*p; i<=n; i+=p) {
                tab[i] = true;
            }
        }
    }
    return tab;
}
int main() {
    int n = 53;
    bool* tab = sieve_of_eratosthenes(n);
    for(int i=2; i<=n; i++) {
        if(!tab[i]){
            cout<<i<<endl;
        }  
    }
}

File: sieve of eratosthenes/python/sieve_of_eratosthenes.py
def sieve_of_eratosthenes(n): #I'll make this return an already marked list up to n
    primes = [True for i in range(n+1)]
    p = 2
    while p*p<=n:
        if primes[p]:
            for j in range(p*p, n+1, p):
                primes[j] = False
        p+=1
    return primes

n = 53
primes = sieve_of_eratosthenes(n)
for i in range(2, n+1):
    if primes[i]:
        print(i)